### PR TITLE
Check service on specification

### PR DESCRIPTION
**Pull request by VLiamin**
Specification - https://docs.google.com/document/d/1EN2_aVLFzQ09Fc2rJCvaHJ77I8ssE0VM1SNL72DEqXI/edit#

### CODE REVIEW

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          EndProject = table.Column<DateTime>(nullable: true),
+          CreatedBy = table.Column<Guid>(nullable: false),
+          CreatedAtUtc = table.Column<DateTime>(nullable: false),
+          ModifiedBy = table.Column<Guid?>(nullable: true),
+          ModifiedAtUtc = table.Column<DateTime?>(nullable: true)
+        },
+        constraints: table =>
+        {
+          table.PrimaryKey("PK_Projects", x => x.Id);
+          table.UniqueConstraint("UX_Project_Name_Unique", x => x.Name);
```
Comment (MariyaSin) [inline src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs:36]: i am not sure

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    private void AddProjectImagesTable(MigrationBuilder migrationBuilder)
+    {
+      migrationBuilder.CreateTable(
+        name: DbProjectImage.TableName,
+        columns: table => new
+        {
+          Id = table.Column<Guid>(nullable: false),
+          EntityId = table.Column<Guid>(nullable: false),
```
Comment (MariyaSin) [inline src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs:85]: ProjectId

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          ImageId = table.Column<Guid>(nullable: false)
+        },
+        constraints: table =>
+        {
+          table.PrimaryKey("PK_EntitiesImages", x => x.Id);
```
Comment (MariyaSin) [inline src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs:90]: ProjectsImages

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs (hunk 1 above)
Comment (VLiamin) [reply src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs]: done

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs (hunk 2 above)
Comment (VLiamin) [reply src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs]: done

File: src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs (hunk 3 above)
Comment (VLiamin) [reply src/ProjectService.Data.Provider.MsSql.Ef/Migrations/20220321165600_InitialTables.cs]: done

File: src/ProjectService.Validation/Project/EditProjectRequestValidator.cs [hunk 4]
```diff
@@ -97,6 +103,18 @@ private async Task HandleInternalPropertyValidationAsync(Operation<EditProjectRe
...
       #endregion
+
+      #region Customer
+
+      AddFailureForPropertyIf(
+        nameof(EditProjectRequest.Customer),
+        x => x == OperationType.Replace,
+        new Dictionary<Func<Operation<EditProjectRequest>, bool>, string>
+        {
+          { x => x.value == null || x.value.ToString().Trim().Length < 150, "Customer is too long." },
```
Comment (MariyaSin) [inline src/ProjectService.Validation/Project/EditProjectRequestValidator.cs:114]: 151

**Review by MariyaSin — APPROVED** [review]

File: src/ProjectService.Validation/Project/EditProjectRequestValidator.cs (hunk 4 above)
Comment (VLiamin) [reply src/ProjectService.Validation/Project/EditProjectRequestValidator.cs]: done

**Review by borovHokage — APPROVED** [review]

**Pull request merged**